pkg/model/network.go

```diff
@@ -87,9 +86,15 @@ func (n NetworkConfig) Disabled() bool {
 
 // IsValid returns an error if any of the fields do not pass validation, or nil
 // otherwise.
-func (n NetworkConfig) IsValid() (err error) {
+func (n NetworkConfig) IsValid() error {
 	if n.Type < NetworkNone || n.Type > NetworkHTTP {
-		err = multierr.Append(err, fmt.Errorf("invalid networking type %q", n.Type))
+		return fmt.Errorf("invalid networking type %q", n.Type)
+	}
+
+	// If the network is enabled, but no domains are specified, the NetworkConfig is considered invalid
+	if !n.Disabled() && len(n.DomainSet()) == 0 {
```

Comment:
We only need domains for HTTP jobs currently – full networking does not enforce an allowlist.

Reply:
👍 addressed in latest commit.